### DISCUSSION TITLE
AWS SQS error when sending message.

### DISCUSSION THREAD

**Discussion by LeifEirik** [landgraf-dev/aws-sdk-delphi#15] [Q&A]
Hi all, 

I'm getting an error when sending a AWS SQS message. The error I get is:
**{
"error": {
"code": "AmazonClientException",
"message": "No RegionEndpoint or ServiceURL configured"
}
}**

My code for sending the message:
```
  const
    AWS_ACCESS_KEY = 'My_AWS_ACCESS_KEY';
    AWS_SECRET_KEY = 'My_AWS_SECRET_KEY';
    AWS_MT_QUEUE   = 'https://sqs.us-east-1.amazonaws.com/MyQueue/ICCMT.fifo';

  procedure SendSQS;
  begin
    lClient := TAmazonSQSClient.Create(AWS_ACCESS_KEY, AWS_SECRET_KEY);
    lClient.Config.RegionEndpoint := TRegionEndpoints.USEast1;

    lMessageRequest := TSendMessageRequest.Create;
    lMessageRequest.MessageBody := '{"client_message_id":12345,"imei":"123456789012345","message":"0123456789ABCDEF"}';
    lMessageRequest.QueueUrl := AWS_MT_QUEUE;
    lMessageRequest.MessageGroupId := TGUID.NewGuid.ToString;
    lMessageRequest.MessageDeduplicationId := TGUID.NewGuid.ToString;

    lMessageResponse := lClient.SendMessage(lMessageRequest);
  end;

```

PS. When I run the code from my development enviroment on a seperate VMWARE machine, the code actually works just fine and sends the messsage as expected. But when I move my executable over to my Windows server on my production enviroment, it fails with the error message mentioned above.

Any hint or help greatly appricated.

-Leif Eirik

**Comment by fastbike** [comment]
Is it some type of firewall issue ?

**Comment by wlandgraf** [comment]
Try this:

```delphi
    lClient := TAmazonSQSClient.Create(AWS_ACCESS_KEY, AWS_SECRET_KEY, TRegionEndpoints.USEast1);
```

And of course, for production environments, never put access key and secret key directly in code.

**Reply by LeifEirik** [reply]
Thank you, that worked!

Not sure though if I understand the difference between the none working code:
```
  lClient := TAmazonSQSClient.Create(AWS_ACCESS_KEY, AWS_SECRET_KEY);
  lClient.Config.RegionEndpoint := TRegionEndpoints.USEast1;
```
...and the code that works:
```
  lClient := TAmazonSQSClient.Create(AWS_ACCESS_KEY, AWS_SECRET_KEY, TRegionEndpoints.USEast1);
```

**Comment by wlandgraf** [comment]
Config is immutable after the client is created. If you want a complex config, you should create the config and then pass it in the constructor of the client.
